libgalois/include/katana/gstl.h

```diff
@@ -40,51 +40,134 @@ namespace katana {
 
 namespace gstl {
 
-//! [define Pow2VarSizeAlloc]
+//! [STL vector using katana Pow2BlockAllocator]
+//! specializes std::vector to use katana concurrent, scaleable allocator:
+//! the allocator is composed of thread-local allocators
+//! that manage thread-local pages and only use a global lock to
+//! get allocation in chunks of (huge) pages.
+//! When destroyed, its memory is reclaimed by the thread that destroys it.
+//!
+//! Use this when allocations and deallocations can occur in a parallel region.
+//! As the memory allocated can be reused for another allocation only by the
+//! thread that deallocated it, this is not suitable for use cases where the
+//! main thread always does the deallocation (after the parallel region).
+//!
+//! If the allocation is large and of known size, then check katana::NumaArray.
+//! If the allocation needs to be uninitialized, then check katana::PodVector.
+//! Read CONTRIBUTING.md for a more detailed comparison between these types.
 template <typename T>
-using Pow2Alloc = Pow2BlockAllocator<T>;
-//! [define Pow2VarSizeAlloc]
-
-template <typename T>
-using FixedSizeAlloc = FixedSizeAllocator<T>;
-
-//! [STL vector using Pow2VarSizeAlloc]
+using Vector = std::vector<T, Pow2BlockAllocator<T>>;
+
+//! [STL deque using katana Pow2BlockAllocator]
+//! specializes std::deque to use katana concurrent, scaleable allocator:
+//! the allocator is composed of thread-local allocators
+//! that manage thread-local pages and only use a global lock to
+//! get allocation in chunks of (huge) pages.
+//! When destroyed, its memory is reclaimed by the thread that destroys it.
+//!
+//! Use this when allocations and deallocations can occur in a parallel region.
+//! As the memory allocated can be reused for another allocation only by the
+//! thread that deallocated it, this is not suitable for use cases where the
+//! main thread always does the deallocation (after the parallel region).
 template <typename T>
-using Vector = std::vector<T, Pow2Alloc<T>>;
-//! [STL vector using Pow2VarSizeAlloc]
-
+using Deque = std::deque<T, Pow2BlockAllocator<T>>;
+
+//! [STL list using katana FixedSizeAllocator]
+//! specializes std::list to use katana concurrent, scaleable allocator:
```

Comment:
The language for List and other containers that use FixedSizeAllocator should read:

The allocator is composed of thread-local allocators that allocate by dividing pages into fixed size blocks. Each per-thread allocator maintain a pool of free pages it replenishes using mmap(), acquiring a global lock.

Reply:
I'm changing it to "the allocator is composed of thread-local allocators that allocate in multiples of (huge) pages by acquiring a global lock and divide the pages into fixed size blocks. Each per-thread allocator maintains a pool of free blocks. When the object is deallocated, its block of memory is added to the free pool of the thread that deallocated it."

Reply:
I'm changing it to "the allocator is composed of thread-local allocators that allocate in multiples of (huge) pages by acquiring a global lock and divide the pages into fixed size blocks. Each per-thread allocator maintains a pool of free blocks. When an object is allocated, it gets a block from the pool of the thread that allocated it. When an object is deallocated, its block of memory is added to the pool of the thread that deallocated it."

Reply:
I'm happy with this text.

Reply:
For Pow2BlockAllocator, I'm changing it to "the allocator is composed of thread-local allocators that allocate in multiples of (huge) pages by acquiring a global lock, and divide the pages into a power-of-2 size blocks. Each per-thread allocator maintains a pool of free blocks of different power-of-2 sizes. When an object is allocated, it gets a block from the pool of the thread that allocated it. When an object is deallocated, its block of memory is added to the pool of the thread that deallocated it."